scripts/globals/abilities/spirit_link.lua

```diff
@@ -70,8 +71,27 @@ function onUseAbility(player,target,ability)
         removeSleepEffects(pet)
     end
 
+    local empathy = player:getMerit(dsp.merit.EMPATHY)
+    if (empathy > 0) then
+        local effects = utils.shuffle(player:getStatusEffects())
```

Comment:
the shuffle might not be needed if we iterate `in pairs` www.luafaq.org/#T1.10

Reply:
I'm only marginally familiar with Lua. Does "in no particular order" imply random? Will multiple consistently yield different results?

Reply:
Using pairs to iterate over a non-contiguous-integer indexed table is not reliable for producing the as-written-in-code order, but it's also not exactly random, so I'd stick with Utils.shuffle.

Interesting side note: the reason this happens is because string-indexed LUA tables use hashes of the strings to order the table, and LUA now randomizes the seed on every new program run to avoid [hash vulnerability](http://lua-users.org/lists/lua-l/2012-01/msg00497.html) that used to exist.  Side effect: your string-indexed tables get mixed around every run.  But within a run, they'll always be ordered the same.